packages/typescript-estree/src/parseSettings/createParseSettings.ts

```diff
@@ -223,3 +228,19 @@ function enforceCodeString(code: unknown): string {
 function getFileName(jsx?: boolean): string {
   return jsx ? 'estree.tsx' : 'estree.ts';
 }
+
+function populateProjectService(
+  optionsRaw: ProjectServiceOptions | true | undefined,
+  settings: CreateProjectServiceSettings,
+) {
+  const options = typeof optionsRaw === 'object' ? optionsRaw : {};
+
+  validateDefaultProjectForFilesGlob(options.allowDefaultProject);
```

Comment:
I created this `populateProjectService` function here so that it could run `validateDefaultProjectForFilesGlob` instead of `createProjectService`. This way we don't need to also move `validateDefaultProjectForFilesGlob` into the new `@typescript-eslint/project-service` package.